### DISCUSSION TITLE
Why ohmybash doesn't use the original code of .bashrc in the new .bashrc?

### DISCUSSION THREAD

**Discussion by ismendoza** [ohmybash/oh-my-bash#440] [General]
Ohmybash backup the .bashrc but the file contains code, what does that code do?
Why the original code is not inside the new .bashrc created by ohmybash?
Why I must trust in a .bashrc without the code created by the operating system?  in my case I use fedora

thank you.

**Comment by akinomyoga** [comment]
> Ohmybash backup the .bashrc but the file contains code, what does that code do?

Which code? The code in the backed-up code? Then, I don't know because I don't have access to your computer. If you don't understand it by reading it by yourself, it probably doesn't do anything that is relevant to you. If you just use the default of Fedora, it sources `/etc/bashrc`, sets `PATH` for user directories, and sources other additional files that you could have created.

> Why the original code is not inside the new .bashrc created by ohmybash?

Because the prompt settings and other hooks can possibly interfere with the existing configurations. It is technically hard to ensure that there are no conflicts with an arbitrary content of an existing .bashrc when we combine both. When the user wants to pick existing settings, the user needs to do it *manually* by carefully thinking about possible conflicts.

> Why I must trust in a .bashrc without the code created by the operating system? in my case I use fedora

What do you mean? Do you mean you do not trust a plain Bash started with an empty .bashrc, where the configurations supplied by the distribution are not sourced? Or do you mean you do not trust any codes added by Oh My Bash? Anyway, you cannot use Oh My Bash if you do not want to trust them, but that is not a bad thing. You just don't have to use Oh My Bash in that case.
